var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -225,6 +225,10 @@ class Gcc(AutotoolsPackage, GNUMirrorPackage):
     # Binutils can't build ld on macOS
     conflicts('+binutils', when='platform=darwin')
 
+    # aarch64/M1 is supported in GCC 12+
+    conflicts('@:11.99', when='target=aarch64: platform=darwin',
```

Comment:
Conflict does not trigger for `darwin-bigsur-m1` yet...

Reply:
I only know of cached package.py within an environment, sorry :(

Also I think you can write this as `conflicts('@:11' ...)`. (The `.99` and `.999` and such scattered throughout spack are horrible.)

Reply:
Oh, I am in an environment indeed. Where do we cache those?

Reply:
Outside an env now.

This successfully triggers the conflict:
* `conflicts('@:11', when='platform=darwin')`

Those should as well but don't:
* `conflicts('@:11', when='target=aarch64: platform=darwin')`
* `conflicts('@:11', when='target=m1 platform=darwin')`
* `conflicts('@:11', when='target=m1: platform=darwin')`

Reply:
@tgamblin is that an arch spec issue?

Reply:
@ax3l Can you please open an issue reporting the concretization behavior above and assign to me?

Reply:
Thanks @alalazo!
It's already open in #26317 :)

Reply:
Oh, you mean the problem that I cannot express the conflict?

Reply:
@alalazo opened in #26520 :) Thank you for taking a look! :)